### PR TITLE
Adds pagination to Inbox view

### PR DESCRIPTION
**Pull request by l2succes**
Just made a PR so you can see the changes<hr data-danger="yep"/>

### Tested on Device?

- [ ] @l2succes

<details>
  <summary>How to get set up with this PR?</summary>
  <p>&nbsp;</p>
   <p><b>To run on your computer:</b></p>
<pre><code>git fetch origin pull/580/head:l2succes-580-checkout
git checkout l2succes-580-checkout
yarn install
cd example; pod install; cd ..
open -a Simulator
yarn start</code></pre>
   </p>
   <p>Then run <code>xcrun simctl launch booted net.artsy.Emission</code> once a the simulator has finished booting</p>
   <p><b>To run inside Eigen (prod or beta) or Emission (beta):</b> Shake the phone to get the Admin menu.</p>
   <p>If you see <i>"Use Staging React Env" </i> - click that and restart, then follow the next step.</p>
   <p>Click on <i>"Choose an RN build" </i> - then pick the one that says: "X,Y,Z"</p>
   <p>Note: this is a TODO for PRs, currently  you can only do it on master commits.</p>
</details>

### CODE REVIEW

**Comment by sarahscott** [comment]
👏

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/components/messages/inbox/__tests__/conversation_snippet-tests.tsx`
- `src/lib/components/messages/inbox/__tests__/inbox-tests.tsx`
- `src/lib/containers/__tests__/messages-tests.tsx`
- `src/lib/relay/__tests__/config-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Could not find stories using these components:

- `ConversationSnippet`
- `Inbox`


  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/components/messages/inbox/inbox.tsx [hunk 1]
```diff
@@ -2,25 +2,85 @@ import * as React from "react"
...
+    }
+  }
+
+  componentDidMount() {
+    this.setState({
+      dataSource: this.state.dataSource,
+    })
+  }
+
+  get conversations() {
```
Comment (alloy) [inline src/lib/components/messages/inbox/inbox.tsx]: Nice ✨

File: src/lib/components/messages/inbox/conversation_snippet.tsx [hunk 2]
```diff
@@ -122,3 +125,31 @@ const styles = StyleSheet.create<Styles>({
...
+  fragments: {
+    conversation: () => Relay.QL`
+      fragment on ConversationType {
+        id
+        inquiry_id
+        from_name
+        from_email
+        to_name
+        last_message
+        artworks {
```
Comment (alloy) [inline src/lib/components/messages/inbox/conversation_snippet.tsx]: It looks like this query is fetching potentially multiple artworks, but [only 1 gets used](https://github.com/artsy/emission/blob/2a2167a083f430867c73bfd834b041ab12c50fc7/src/lib/components/messages/inbox/conversation_snippet.tsx#L79). Is that correct?

**Pull request merged**